### PR TITLE
Metal driver implementation

### PR DESCRIPTION
**Pull request by bejado**
Basic Metal support. Still a lot of TODOs, but this is enough to get the lightbulb sample working correctly.

### CODE REVIEW

**Comment by prideout** [comment]
🍾

File: shaders/src/post_process.vs [hunk 1]
```diff
@@ -22,5 +22,13 @@ void main() {
...
     vertex_uv = (floor(vertex_uv) + vec2(0.5, 0.5)) * frameUniforms.resolution.zw;
 #endif
+
     gl_Position = position;
+
+#if defined(TARGET_METAL_ENVIRONMENT)
+    // Metal texture space is vertically flipped that of OpenGL's, so flip the Y coords so we sample
+    // the frame correctly. Vulkan doesn't need this fix because its clip space is mirrored
+    // (the Y axis points down the screen).
+    gl_Position.y = -gl_Position.y;
```
Comment (romainguy) [inline shaders/src/post_process.vs]: We'll have to be careful with this when multiple post-process steps are used. Also what happens when post-processing is completely disabled then? Or if FXAA is disabled but the rest of the post-processing is still enabled?

Comment (bejado) [reply shaders/src/post_process.vs]: The vertical flip compensates for Metal’s Y-inverted texture space, putting the UV origin of the fullscreen quad at the top left of the screen (as opposed to the bottom left for OpenGL). It makes no difference whether an even or odd number of post processing passes happen.

This is what the vertex shader for the fullscreen quad looks like:
postprocess_vertex
Either way, the orientation is correct within each pass.

Comment (romainguy) [reply shaders/src/post_process.vs]: Makes sense, thanks!

Comment (romainguy) [reply shaders/src/post_process.vs]: BTW glslang can do this on our behalf, there's a special option for it.

Comment (bejado) [reply shaders/src/post_process.vs]: Yeah, the `flip_vert_y` flag? We only want to do it for the post-processing shaders so `GLSLPostProcessor` would need some logic to determine if it's a material shader or a post-processing one.

**Pull request merged**